debugger/show_source.ml

```diff
@@ -23,18 +23,23 @@ open Source
 
 (* Print a line; return the beginning of the next line *)
 let print_line buffer line_number start point before =
-  let next = next_linefeed buffer start
+  let next_linefeed = next_linefeed buffer start
   and content = buffer_content buffer
   in
     printf "%i " line_number;
+    let next =
+      if content.[next_linefeed - 1] = '\r' then
+        next_linefeed - 1
+      else
+        next_linefeed in
     if point <= next && point >= start then
       (print_string (String.sub content start (point - start));
        print_string (if before then event_mark_before else event_mark_after);
        print_string (String.sub content point (next - point)))
     else
       print_string (String.sub content start (next - start));
     print_newline ();
-    next
+    next_linefeed
```

Comment:
The shadowing was inadvertent! Not renaming `next` was consciously in order to simplify the patch, vs simplifying the code. Would it be clearer if the meaning of `next` were instead commented?